File: 프로그래머스/lv1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    member = [1]*(n+2)
    cnt = 0 
    for i in range(1, n+1):
        if i in lost:
            member[i] -= 1 
        if i in reserve:
            member[i] += 1 
    for i in range(1, n+1):
        if member[i] == 2:
            if member[i-1] == 0:
                member[i-1] += 1 
                member[i] -= 1 
            elif member[i+1] == 0:
                member[i+1] += 1 
                member[i] -= 1 
    for i in range(1, n+1):
        if member[i] >= 1:
            cnt += 1 
    return cnt

File: 프로그래머스/1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {
    // 여분의 체육복이 있지만 도난당한 학생들을 먼저 제거
    let actualReserve = reserve.filter(r => !lost.includes(r));
    let actualLost = lost.filter(l => !reserve.includes(l));

    // 여분의 체육복이 있는 학생들 정렬
    actualReserve.sort((a, b) => a - b);
    actualLost.sort((a, b) => a - b);

    let answer = n - actualLost.length;

    // 체육복을 빌려줄 수 있는지 확인
    actualReserve.forEach(r => {
        let index = actualLost.findIndex(l => Math.abs(l - r) === 1);
        if (index !== -1) {
            answer++;
            actualLost.splice(index, 1); // 빌려준 학생은 목록에서 제거
        }
    });

    return answer;
}
